exchange/src/exchange/wrappers/handle/wrapper_handle.cpp

```diff
@@ -46,17 +48,25 @@ WrapperHandle::~WrapperHandle()
     }
 }
 
-WrapperHandle::WrapperHandle(
-    const std::string& remote_uid, const std::string& algo_id
-) :
-    WrapperHandle({"--uid", quote_id(remote_uid), "--algo_id", quote_id(algo_id)})
-{}
+WrapperHandle::WrapperHandle(const std::string& remote_uid, const std::string& algo_id)
+{
+    std::optional<std::string> remote_algo =
+        nutc::firebase::get_algo(remote_uid, algo_id);
 
-WrapperHandle::WrapperHandle(const std::string& algo_path) :
     WrapperHandle(
```

Comment:
This isn't how you overload constructors. Construct with the function call in the arg position so you don't need to construct inside the brackets